website/docs/cloud-docs/registry/publish-providers.mdx

```diff
@@ -20,72 +20,196 @@ Users must be members of an organization to access its registry and private prov
+    "attributes": {
+      "namespace": "ORG_NAME",
+      "ascii-armor": "-----BEGIN PGP PUBLIC KEY BLOCK-----\n\nmQINB...=txfz\n-----END PGP PUBLIC KEY BLOCK-----\n"
+    }  }
+}
+```
+
+Use the [Add a GPG key endpoint](/terraform/cloud-docs/api-docs/private-registry/gpg-keys#add-a-gpg-key) to add the public key that matches the signing key for the release. Replace `TOKEN` in the `Authorization` header with your Terraform Cloud API token.
+
+```shell-session
+$ curl \
+  --header "Authorization: Bearer TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request POST \
+  --data @key.json \
+  https://app.terraform.io/api/registry/private/v2/gpg-keys
+```
+
+The response contains a `key-id` that you will use to create a provider version.
+
+```json
+"key-id": "34365D9472D7468F"
+```
+
+### Create a version
+
+Create a file named `version.json` with the following contents. Replace the value of the `version` field with the version of your provider, and replace the `key-id` field with the id of the GPG key that you created in the [Add your public key](#add-your-public-key) step.
+
+```hcl
+{
+  "data": {
+    "type": "registry-provider-versions",
+    "attributes": {
+      "version": "5.14.0",
+      "key-id": "34365D9472D7468F",
+      "protocols": ["5.0"]
+    }
+  }
+}
+```
+
+Use the [Create a Provider Version endpoint](/terraform/cloud-docs/api-docs/private-registry/provider-versions-platforms#create-a-provider-version) to create a version for your provider. Replace `TOKEN` in the `Authorization` header with your Terraform Cloud API token and replace both instances of `ORG_NAME` with the name of your organization.
+
+```shell-session
+$ curl \
+  --header "Authorization: Bearer TOKEN" \
+  --header "Content-Type: application/vnd.api+json" \
+  --request POST \
+  --data @payload.json \
+  https://app.terraform.io/api/v2/organizations/ORG_NAME/registry-providers/private/ORG_NAME/aws/versions
+```
+
+ The response includes URL links that you will use to upload the `SHA256SUMS` and `SHA256.sig` files.
+
+```json
+"links": {
+    "shasums-upload":     "https://archivist.terraform.io/v1/object/dmF1b64hd73ghd63",
+    "shasums-sig-upload": "https://archivist.terraform.io/v1/object/dmF1b37dj37dh33d"
+  }
+```
+
+### Upload signatures
+
+Upload the `SHA256SUMS` and `SHA256SUMS.sig` files to the URLs returned in the previous step. The example command below uploads the files from your local machine. First upload the `SHA256SUMS` file to the URL returned in the `shasums-upload` field.
+
+```shell-session
+$ curl \
+  -T terraform-provider-aws_5.14.0_SHA256SUMS \
+  https://archivist.terraform.io/v1/object/dmF1b64hd73ghd63...
+```
+
+Next, upload the `SHA256SUMS.sig` file to the URL returned in the `shasums-sig-upload` field.
+
+```shell-session
+$ curl \
+  -T terraform-provider-aws_5.14.0_SHA256SUMS.72D7468F.sig \
+  https://archivist.terraform.io/v1/object/dmF1b37dj37dh33d...
+```
+
+### Create a provider platform endpoint
```

Comment:
❓ Why did you put `endpoint` onto this heading? We usually refer to them as `provider platforms` or `provider binaries`.

Reply:
Typo :) Nice catch!